diesel/src/mysql/backend.rs

```diff
@@ -64,3 +64,24 @@ impl HasSqlType<::types::Timestamp> for Mysql {
         MysqlType::Timestamp
     }
 }
+
+use types::{ToSql, IsNull, FromSql};
+use std::error::Error as StdError;
+use std::io::Write;
+
+impl ToSql<::types::Bool, Mysql> for bool {
+    fn to_sql<W: Write>(&self, out: &mut W) -> Result<IsNull, Box<StdError+Send+Sync>> {
+        let int_value = if *self {
+            1
+        } else {
+            0
+        };
+        <i32 as ToSql<::types::Integer, Mysql>>::to_sql(&int_value, out)
+    }
+}
+
+impl FromSql<::types::Bool, Mysql> for bool {
+    fn from_sql(bytes: Option<&[u8]>) -> Result<Self, Box<StdError+Send+Sync>> {
+        Ok(not_none!(bytes)[0] != 0)
```

Comment:
Yeah, boolean expressions in MySQL are transmitted as `MYSQL_TYPE_TINY` which corresponds to `signed char`. For codegen I will infer `TINYINT(1)` to mean bool, yes. I don't really want to define/care about the behavior of a > 1 byte type being used here, but I think your code is actually easier to read anyway